### PR TITLE
Small refactoring for `FutureUnodered` iterators

### PR DESCRIPTION
**Pull request by MajorBreakfast**
- Put both iterators into a single file
- Make `iter_mut()` use `iter_pin_mut()` internally
- Add `Unpin` impl for `IterMut` (because why not)

### CODE REVIEW

File: futures-util/src/stream/futures_unordered/iter.rs [hunk 1]
```diff
@@ -12,6 +12,10 @@ pub struct IterPinMut<'a, F: 'a> {
     pub(super) _marker: PhantomData<&'a mut FuturesUnordered<F>>
 }
 
+#[derive(Debug)]
+/// Mutable iterator over all futures in the unordered set.
+pub struct IterMut<'a, F: 'a + Unpin> (pub(super) IterPinMut<'a, F>);
+
 impl<'a, F: Unpin> Unpin for IterPinMut<'a, F> {}
```
Comment (cramertj) [inline futures-util/src/stream/futures_unordered/iter.rs]: `IterPinMut` is `Unpin` whether or not `F` is `Unpin`, right?

Comment (MajorBreakfast) [reply futures-util/src/stream/futures_unordered/iter.rs]: Yes because it doesn't store the data. I've removed this restriction

Comment (MajorBreakfast) [reply futures-util/src/stream/futures_unordered/iter.rs]: Should I remove both `Unpin` impls? `Bilock` and other types that don't use pinning typically don't implement `Unpin`

Comment (MajorBreakfast) [reply futures-util/src/stream/futures_unordered/iter.rs]: I've removed them. If you think I should put the back I can put them back in, but I don't think these impls are useful.

**Comment by cramertj** [comment]
LGTM! Thanks for cleaning this up!

**Pull request merged**